.github/workflows/lint-compile-verify.yml

```diff
@@ -0,0 +1,111 @@
+name: lint, compile and build
+on:
+  push:
+    branches: main
+  pull_request:
+
+jobs:
+  lint-env-and-nf:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: jidicula/clang-format-action@v4.9.0
+
+  # TODO lint-tool
+
+  # TODO document-env
+
+  compile-nf:
+    needs: lint-env-and-nf
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # keep list in sync with build-nf
+        nf:
+          - bpf
+          - bridge
+          - dhcp_ipv4
+          - dns_aaa_records
+          - firewall
+          - iptables
+          - iptables_blacklist
+          - iptables_port_whitelist
+          - iptables_whitelist
+          - maglev
+          - nat
+          - nop
+          - policer
+          - router
+          - rust-policer
+        exclude:
+          # TODO fix makefile
+          - nf: bpf
```

Comment:
This can be a permanent exclude, the bpf NFs were really only examples for the paper